modules/oe_editorial_entity_version/src/Form/NodeRevisionRevertForm.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_editorial_entity_version\Form;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Datetime\DateFormatterInterface;
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\RevisionableInterface;
+use Drupal\Core\Form\ConfirmFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a form for reverting a node revision.
+ */
+class NodeRevisionRevertForm extends ConfirmFormBase {
```

Comment:
I think you should extend \Drupal\entity_version_workflows\Form\NodeRevisionRevertForm which already extends the core form and just override the methods where changes are needed.

Reply:
Yes, I started like that, but I ended up overriding 90% of that class so that does not make sense anymore to extend it and override everything.